core/server/models/user.js

```diff
@@ -451,14 +454,16 @@ User = ghostBookshelf.Model.extend({
             // Assign the hashed password
             userData.password = hash;
 
-            return Promise.join(
-                gravatar.lookup(userData),
-                ghostBookshelf.Model.generateSlug.call(this, User, userData.name, options)
-            );
-        }).then(function then(results) {
-            userData = results[0];
-            userData.slug = results[1];
+            return gravatar.lookup(userData)
```

Comment:
Do these calls really need to be done one-after-the other? 
Maybe a candidate for [promise.props](http://bluebirdjs.com/docs/api/promise.props.html) ?